arangod/Transaction/Methods.h

```diff
@@ -206,20 +206,20 @@ class Methods {
   Result begin();
 
   /// @deprecated use async variant
-  Result commit();
+  auto commit() noexcept -> Result;
   /// @brief commit / finish the transaction
-  Future<Result> commitAsync();
+  auto commitAsync() noexcept -> Future<Result>;
 
   /// @deprecated use async variant
-  Result abort();
+  auto abort() noexcept -> Result;
   /// @brief abort the transaction
-  Future<Result> abortAsync();
+  auto abortAsync() noexcept -> Future<Result>;
 
   /// @deprecated use async variant
-  Result finish(Result const& res);
+  auto finish(Result const& res) noexcept -> Result;
 
   /// @brief finish a transaction (commit or abort), based on the previous state
-  Future<Result> finishAsync(Result const& res);
+  auto finishAsync(Result const& res) noexcept -> Future<Result>;
```

Comment:
do we want to add `[[nodiscard]]` to these functions?

Reply:
(the return values are discarded all over the place, so I decided to put it into a separate PR)